### PR TITLE
Increasing Data Schema's Scope to include sd.yaml and validator IO

### PR DESCRIPTION
**Pull request by stjohnjohnson**
- Major version bump for backwards incompatible changes
- index now exposes access to config, api, and model
- Added examples for all config and api validation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,21 +7,30 @@
...
  - `update` - Expected input values when making an UPDATE action against this resource
+ - `keys` - List of keys that combine to represent a unique row
+ - `tableName` - Internal name of the table
+ - `indexes` - Secondary indexes to make search/lookup faster
 
 ## Usage
 
 ```bash
-npm install screwdriver-data-model
+npm install screwdriver-data-schema
```
Comment (tkyi) [inline README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -7,21 +7,30 @@
...
+ - `models` - Internal data resources
 
-## Methodology
+## Models
+
+The model represents a combination of what is required to create the resource and what is possible for returning from the resource.
+
+### Methodology
 
 Each model contains four (4) schemas:
```
Comment (nkatzman) [inline README.md]: 7 != 4

Comment (d2lam) [inline README.md]: Change this number

File: config/workflow.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const Joi = require('joi');
+const Regex = require('./regex');
+
+const SCHEMA_WORKFLOW = Joi.array()
```
Comment (nkatzman) [inline config/workflow.js]: Would it be possible to set 'main' as the first field here by default?

Comment (nkatzman) [inline config/workflow.js]: Or is that a plan to do inside the config-parser?

**Comment by nkatzman** [comment]
Besides my small comments, 👍

File: config/workflow.js (hunk 3 above)
Comment (stjohnjohnson) [inline config/workflow.js]: Nah, main doesn't belong in Workflow.

**Comment by d2lam** [comment]
👍

**Pull request merged**